### PR TITLE
Fix local-on bug

### PR DESCRIPTION
**Pull request by benharsh**
Local-on statements were incorrectly created in cases like these:
```chpl
local local on here do writeln("hi");

local {
  on here do writeln("hi");
}
```

Both of those should be local-statements with some kind of on-statement inside them, but were turned into local-on statements. By using the utility function 'findStmtWithTag', I was simply checking to see if the inner statement was an on-statement. Instead, I should check if the stmt is scopeless and is a regular on-statement (not a local-on).

Credit to @cassella for discovering this issue.

### CODE REVIEW

**Comment by benharsh** [comment]
I still have some testing to do:
- [ ] local-numa
- [ ] gasnet numa

File: test/optimizations/localon/local-block.good [hunk 1]
```diff
@@ -0,0 +1,2 @@
+{x = 5}
+{x = local-block.chpl:12: error: cannot access remote data in local block
```
Comment (ronawho) [inline test/optimizations/localon/local-block.good]: Skipif or suppessif for --fast?

Comment (ronawho) [inline test/optimizations/localon/local-block.good]: Oh wait, I guess you're only running it for comm!=none, so you don't need to suppress --fast

**Comment by ronawho** [comment]
looks good to me!

**Pull request merged**